### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by lucas-crodrigues**
This is part 1 of the School Library app initiated in the third week, of the fourth module in the Microverse program.

The requirements are:

> ### Project requirements
> - Create class `Person` with  the following:
>   - Instance vars: `@id`, `@name`, and `@age`.
>   - Constructor with `name`,  `age`, and `parent_permission` as parameter. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
>   - Getters for `@id`, `@name`, and `@age`.
>   - Setters for `@name` and `@age`.
>   - Private method `is_of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
>   - Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.
> - Create class `Student` with the following:
>   - Inherits from `Person`.
>   - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
>   - Method `play_hooky` that returns `"¯\(ツ)/¯"`.
> - Create class `Teacher` with the following:
>   - Inherits from `Person`.
>   - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
>   - Override `can_use_services?` so it always returns `true`.
> - Each class should be saved in a separate file.

- [x] Create class `Person`
- [x] Create class `Student` 
- [x] Create class `Teacher` 

Thank you

### CODE REVIEW

File: teacher.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(specialization, age, name = 'Unknown')
+    super(age, name)
```
Comment (sja-thedude) [inline teacher.rb]: - [x] Kindly add parent permission as parameter to initialize and super constructors as its a requirement.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @age = age
```
Comment (sja-thedude) [inline person.rb:7]: - [x] Kindly make sure to add `@id = Random.rand(1..1000)` along with age, name and parent permission as its the requirement.

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
```
Comment (sja-thedude) [inline person.rb:5]: - [x] Kindly make sure to add getter and setter methods before initialize constructor as below:
```ruby
attr_accessor :name, :age 
  attr_reader :id 
```

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @lucas-crodrigues, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Good job so far
✔️ Professional Readme
✔️ Correct Gitflow

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb (hunk 1 above)
Comment (lucas-crodrigues) [reply teacher.rb]: Will do, thank you

File: person.rb (hunk 2 above)
Comment (lucas-crodrigues) [reply person.rb:7]: Went completely over my head, thanks for pointing it out

File: person.rb (hunk 3 above)
Comment (lucas-crodrigues) [reply person.rb:5]: Thank you for the feedback

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @lucas-crodrigues , 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive PR.
:heavy_check_mark: Well implemented suggested changes.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "fix bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**